### ISSUE TITLE
False positive on missing region for endregion comment

### ISSUE THREAD

**Issue by ItsDrike** [microsoft/pyright#4496]
**Describe the bug**
After an update from 1.1.286 to 1.1.290, I'm experiencing a bug that falsly marks any `# endregion` comment as not having a matching `# region` , even though it is present.

**To Reproduce**
Create file with the following content:
```python
# region: foo

x = 5

# endregion
```

Run pyright and notice an error on line 4, stating that endregion is missing a region.

**Expected behavior**
Pyright should be able to find the region tag above properly, and not report this error.

**Screenshots or Code**
![image](https://user-images.githubusercontent.com/20902250/213640833-167fa72b-2cac-4524-9570-4d888b63d716.png)

**Comment by erictraut** [comment]
@debonte, any thoughts on this one?

**Comment by ItsDrike** [comment]
Weirdly enough, doing 
```python
# region: foo

x = 5

# endregion: anything
``` works just fine and doesn't produce the error

**Comment by debonte** [comment]
We changed how we decide if a comment is a region/endregion for this issue: https://github.com/microsoft/pylance-release/issues/3765.

Seems like that may need some tweaking still.

**Comment by debonte** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by rchl** [comment]
I'll just add that this error has also triggered in a code that just had a comment that happened to start with a "region word.

```
            for diagnostic, candidate in diagnostics:
                # Checking against points is inclusive unlike checking whether region intersects another
                # region which is exclusive (at region end) and we want an inclusive behavior in this case.
```

It's obviously easy to workaround by changing the position of the "region" word but just wanted to point it out, in case this is not supposed to trigger the error.

I'm not sure if there is any spec for how those markers should work but I guess the match mattern could be a bit stricter and require the comment to either end after `# region` or be directly followed by semicolon.

**Comment by erictraut** [comment]
This is included in pyright 1.1.291, which I just published. It will also be included in a future release of pylance.
